### ISSUE TITLE
BBB cluster using ScaleLite with Nginx-proxy

### ISSUE THREAD

**Issue by IamLuksha** [bigbluebutton/bigbluebutton#9063]
Hello!

We are trying to deploy a BBB cluster using ScaleLite. Now each BBB server has an external ip address. But the number of servers is not enough for us now. It is necessary to increase their number. I would like to use the least number of external ip addresses.

The diagram below shows an example of what we want to do.
Let the external ip-addresses 20.20.20. * And the internal 172.16.2. *

Using
https://docs.bigbluebutton.org/2.2/setup-turn-server.html
https://docs.bigbluebutton.org/2.2/configure-firewall.html
https://github.com/bigbluebutton/bbb-install/issues/133
and other...

Nothing succeeded. TCP packets work, UDP not.

Please help configure BBB for such a scheme.
And is it even possible?
4 days trying to configure but nothing works. Web works, but no voice, webcams and screenshare.
In short, "Error during WebSocket handshake: Unexpected response code: 400"

Ready to report any current settings that are needed for help

![Untitled Diagram (1)](https://user-images.githubusercontent.com/18416207/79028351-52a4d100-7b98-11ea-95e3-1e8dfb277631.png)

**Comment by tomlp** [comment]
Dear Luksha,
i think it´s not possible with the current setup method. Every BBB-Server has it´s own Freeswitch and it needs UDP-Ports in his nature for direct access. As you can see in the config files of freeswitch, it must have IP-addresses entered and no DNS-Names ("it will not work" is printed in the config file).
DNS-Names on this places would give the option to use "sslh" to redirect - based on the DNS-Name - to the correct FreeSwitch... (of course: nginx has also the ability to redirect streams)
A other Option would be a dedicated Freeswitch(-Cluster), that holds all Audio-Streams. But the config-files have all to be adapted (and there are a lot of places where the config entries has to be changed :( )
Kurento should work (in my opinion) with DNS-Names, so there should be redirected over nginx/sslh.
May be the developer of BBB had thought about the scalability of BBB, but for now i think its not easy done :/
kr Tom
PS: i have also 2 Clusters of BBB. one with 3 Nodes and a second with 2 nodes.... and had the same problem in the past. my solution: i buy more IP-addresses :) 
PPS : May be the **change to IPv6** would help ?!? since 6. April 2019 there is the commitment that all components supports IPv6, and nearly all providers give IPv6 Addresses in masses ....

**Comment by IamLuksha** [comment]
Thanks for the answer!

I am using DNS Names.

Now there are free IP addresses purchased earlier. Now they have given each BBB server their external IP address. But to solve this problem would be interesting.

But for now we got a new problem

https://github.com/blindsidenetworks/scalelite/issues/148

**Comment by tomlp** [comment]
strange - sorry, i use Greenlight as frontend (in Austria, with äöü too) but had never (until now) this kind of problem....
kr Tom

**Comment by matiasilva** [comment]
Closed due to inactivity -- seems like the problem has been solved.
